test/ruby/test_regexp.rb

```diff
@@ -1827,6 +1827,17 @@ def test_s_timeout_memory_leak
     end;
   end
 
+  def test_bug_20453
+    assert_separately([], "#{<<-"begin;"}\n#{<<-'end;'}")
+    begin;
+      Regexp.timeout = 0.001
+
+      assert_raise(Regexp::TimeoutError) do
+        /^(a*)x$/ =~ "a" * 1000000 + "x"
+      end
```

Comment:
It is possible without changing the global setting.
And `assert_separately` may not be necessary, if you use it for this purpose.

```suggestion
      re = Regexp.new("^(a*)x$", timeout: 0.001)

      assert_raise(Regexp::TimeoutError) do
        re =~ "a" * 1000000 + "x"
      end
```

Reply:
Yeah, that makes good sense to me. Thank you!

Reply:
Oops, I merged it before you were able to address this comment. Can you open a followup PR that addresses this?